classes/Kohana/Jam/Model.php

```diff
@@ -236,6 +236,8 @@ public function update_fields($values, $value = NULL)
 	 *
 	 * @param   bool|null        $validate
 	 * @return  $this
+	 * @throws Kohana_Exception
+	 * @throws Jam_Exception_Validation
```

Comment:
Shouldn't this annotation only be on `check_insist()`?

Reply:
Jam_Exception_Validation can be added to check_insist() as well but it must be present in save() as well.
I did not go deeper, check_insist() is used mostly in jam model except for two other occupancies. do you suggest that we put the annotation in  check_insist() as well

Reply:
IMHO, `@throws` annotation should be put only on methods which actually the respective `throw` call. It shouldn't describe every possible nested `throw` in the call stack. If we were to follow that logic controllers should be denoted with `@throws` for most of the possible exceptions.

Reply:
So yes, `check_insist()` _could_ have that annotation, but `save()` _must not_ have it.

---

I think much of these `@throws` annotations are caused by JetBrains IDEs reporting missing `@throws` annotations for nested calls which I think is incorrect.

Reply:
after discussion we decided to deprecate check_insist()